lib/reverse_words.rb

```diff
@@ -1,6 +1,37 @@
+require "pry"
 # A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+#
+# Time complexity: O(n) - The outer loop is O(n). Each time the inner loop runs, it operates on a small part of the array, so this inner section is O(1). Added together, they add up to:
+# work done by outer loop   +   total work done by inner loop
+#         O(n)              +    n*O(1)
+# The final result evaluates to O(n).
+#
+# Space complexity: O(1)
+
 def reverse_words(my_words)
-  raise NotImplementedError
+  return nil if my_words == nil
+
+  i_start = 0
+  i_end = 0
+
+  until i_start >= (my_words.length - 1)
+    until (my_words[i_end] == " ") || (i_end == my_words.length)
+      i_end += 1
+    end
+
+    i = 0
+    until i == (i_end - i_start) / 2
```

Comment:
Style nit: could we refactor this into a helper method? (Perhaps we can even use a method we wrote for an earlier assignment? 🙂)